.travis.yml

```diff
@@ -0,0 +1,49 @@
+language: julia
+
+os:
+  - linux
+  - osx
+  - freebsd
+  - windows
+
+julia:
+  - 1.0
+  - 1.4
+  - nightly
+
+notifications:
+  email: false
+
+cache:
+  directories:
+    - $HOME/.julia/artifacts
+
+branches:
+  only:
+  - master
+  - gh-pages # For building documentation
+  - /^testing-.*$/ # testing branches
+  - /^v[0-9]+\.[0-9]+\.[0-9]+$/ # version tags
```

Comment:
This feels overspecified and also means no PRs will get run unless they explicitly start with `testing-`. While you are free to impose that restriction on contributors, I think keeping the barrier to contribution as low as possible is key to getting people to help!

Reply:
Pull requests against master are always run. This setting prevents running CI on branches, which saves a lot of CI time, especially when opening pull requests from branches on this repository

Reply:
ah, I didn't know that!